plugins/ua_securitypolicy_none.c

```diff
@@ -0,0 +1,304 @@
+channelContext_setLocalSymEncryptingKey_sp_none(void *const channelContext,
+                                                const UA_ByteString *const key) {
+    UA_SP_NONE_ChannelContextData* const data = (UA_SP_NONE_ChannelContextData*)channelContext;
+    data->callCounter++;
+    return UA_STATUSCODE_GOOD;
+}
+
+static UA_StatusCode
+channelContext_setLocalSymSigningKey_sp_none(void *const channelContext,
+                                             const UA_ByteString *const key) {
+    UA_SP_NONE_ChannelContextData* const data = (UA_SP_NONE_ChannelContextData*)channelContext;
+    data->callCounter++;
+    return UA_STATUSCODE_GOOD;
+}
+
+
+static UA_StatusCode
+channelContext_setLocalSymIv_sp_none(void *const channelContext,
+                                     const UA_ByteString *const iv) {
+    UA_SP_NONE_ChannelContextData* const data = (UA_SP_NONE_ChannelContextData*)channelContext;
+    data->callCounter++;
+    return UA_STATUSCODE_GOOD;
+}
+
+static UA_StatusCode
+channelContext_setRemoteSymEncryptingKey_sp_none(void *const channelContext,
+                                                 const UA_ByteString *const key) {
+    UA_SP_NONE_ChannelContextData* const data = (UA_SP_NONE_ChannelContextData*)channelContext;
+    data->callCounter++;
+    return UA_STATUSCODE_GOOD;
+}
+
+static UA_StatusCode
+channelContext_setRemoteSymSigningKey_sp_none(void *const channelContext,
+                                              const UA_ByteString *const key) {
+    UA_SP_NONE_ChannelContextData* const data = (UA_SP_NONE_ChannelContextData*)channelContext;
+    data->callCounter++;
+    return UA_STATUSCODE_GOOD;
+}
+
+static UA_StatusCode
+channelContext_setRemoteSymIv_sp_none(void *const channelContext,
+                                      const UA_ByteString *const iv) {
+    UA_SP_NONE_ChannelContextData* const data = (UA_SP_NONE_ChannelContextData*)channelContext;
+    data->callCounter++;
+    return UA_STATUSCODE_GOOD;
+}
+
+static UA_StatusCode
+channelContext_compareCertificate_sp_none(const void *const channelContext,
+                                          const UA_ByteString *const certificate) {
+    if(channelContext == NULL || certificate == NULL)
+        return UA_STATUSCODE_BADINTERNALERROR;
+
+    return UA_STATUSCODE_GOOD;
+}
+
+static size_t
+channelContext_getRemoteAsymPlainTextBlockSize_sp_none(const void *const channelContext) {
+    return 0;
+}
+
+static size_t
+channelContext_getRemoteAsymEncryptionBufferLengthOverhead_sp_none(const void *const channelContext,
+                                                                   const size_t maxEncryptionLength) {
+    return 0;
+}
+
+/*****************************/
+/* Security Policy functions */
+/*****************************/
+
+static void
+policy_deletemembers_sp_none(UA_SecurityPolicy *policy) {
+    UA_ByteString_deleteMembers(&policy->localCertificate);
+}
+
+UA_StatusCode
+UA_SecurityPolicy_None(UA_SecurityPolicy *policy,

```

Comment:
I like that we create the policy this way now, and that the context is inside of the policy.
Although the optional argument is now missing. If the policy needs some other user input besides certificates, trust lists or keys, there is now no way of passing this to the policy(context). For example if the random generator needs random sources which the user must supply. I propose we add a void pointer and make an implementation specific function (something like UA_SecurityPolicy_None_mbedtls). (There might be different implementations of the same policy)